addons-dev/aldryn-addons/aldryn_addons/settings.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+import imp
+import os
+import shutil
+import uuid
+from aldryn_addons import utils
+from pprint import pformat
+from django.core.exceptions import ImproperlyConfigured
+
+
+def save_settings_dump(settings, path):
+    to_dump = {key: value for key, value in settings.items() if key == key.upper()}
+    with open(path, 'w+') as fobj:
+        fobj.write(
+            pformat(
+                to_dump,
+                indent=4,
+            )
+        )
+
+
+def count_str(number):
+    return '{0:05d}'.format(number)
+
+
+def load(settings):
+    settings['BASE_DIR'] = settings.get(
+        'BASE_DIR',
+        os.path.dirname(os.path.abspath(settings['__file__']))
+    )
+    settings['ADDONS_DIR'] = settings.get(
+        'ADDONS_DIR',
+        os.path.join(settings['BASE_DIR'], 'addons')
+    )
+    settings['ADDONS_DEV_DIR'] = settings.get(
+        'ADDONS_DEV_DIR',
+        os.path.join(settings['BASE_DIR'], 'addons-dev')
+    )
+    utils.mkdirs(settings['ADDONS_DIR'])
+    # TODO: .debug is not multi-process safe!
+    debug_path = os.path.join(settings['ADDONS_DIR'], '.debug')
+    shutil.rmtree(debug_path, ignore_errors=True)
```

Comment:
is it necessary to remove everything in .debug every time this is called? Not sure how often it gets called so I ask :)